src/main/java/com/google/maps/PendingResultBase.java

```diff
@@ -96,8 +96,18 @@ protected A param(String key, String val) {
     return result;
   }
 
+  protected A param(String key, int val) {
+    params.put(key, Integer.toString(val));
+
+    @SuppressWarnings("unchecked") // safe by specification - A is the actual class of this instance
```

Comment:
minor nit: could you replace these three returns with

```java
private A getInstance() {
  @SuppressWarnings("unchecked");
  return (A) this;
}

// later
return this.getInstance();
```

Reply:
I like it, done. Ish.